Add failover docs (#3)
aterleto/jedis: README.md

jedis.sadd("planets", "Mars");
```

## Failover

Jedis supports retry and failover for your Redis deployments. This is useful when:
1. You have more than one Redis deployment. This might include two independent Redis servers or two or more Redis databases replicated across multiple [active-active Redis Enterprise](https://docs.redis.com/latest/rs/databases/active-active/) clusters.
2. You want your application to connect to one deployment at a time and to fail over to the next available deployment if the first deployment becomes unavailable.

### Configuring failover

> To configure Jedis for failover, you specify an ordered list of Redis databases to connect to.

Reviewer: ```suggestion
To configure Jedis for failover, you specify an ordered list of Redis databases to connect.
```

Author: Thanks for pointing this one out. I've further revised it in my subsequent review.

> To configure Jedis for failover, you specify an ordered list of Redis databases to connect to.

Author: ```suggestion
To configure Jedis for failover, you specify an ordered list of Redis databases.
```

By default, Jedis will connect to the first Redis database in the list. If the first database becomes unavailable,
Jedis will attempt to connect to the next database in the list, and so on.

What follows here is a simple configuration example. For more details, see the [complete Jedis failover docs](docs/failover.md).

First create an array of `ClusterJedisClientConfig` objects, one for each Redis database.

```java
JedisClientConfig config = DefaultJedisClientConfig.builder().user("cache").password("secret").build();

ClusterJedisClientConfig[] clientConfigs = new ClusterJedisClientConfig[2];
clientConfigs[0] = new ClusterJedisClientConfig(new HostAndPort("redis-east.example.com", 14000), config);
clientConfigs[1] = new ClusterJedisClientConfig(new HostAndPort("redis-west.example.com", 14000), config);
```

The configuration above represents two example Redis databases: `redis-east.example.com` and `redis-west.example.com`.

> (Note: If you're running Redis Enterprise, it's important that these two databases reside in separate clusters.)

Reviewer: ```suggestion
(Note: If you're running Redis Enterprise, it's logical that these two databases would reside in separate clusters that are typically geo-distributed for regional/datacenter failover.)
```

Author: Good suggestion on this. I've slightly word-smithed this. See my review.

> (Note: If you're running Redis Enterprise, it's important that these two databases reside in separate clusters.)

Author: ```suggestion
(Note: If you're running Redis Enterprise, we recommend that these two databases reside in separate clusters that are geo-distributed for regional / data center failover.)
```

Author: @aterleto I slightly revised your suggestion here.

You can use these configuration objects to create and configure a connection provider that supports failover:

```java
MultiClusterJedisClientConfig.Builder builder = new MultiClusterJedisClientConfig.Builder(clientConfigs);
builder.circuitBreakerSlidingWindowSize(10);
builder.circuitBreakerSlidingWindowMinCalls(1);
builder.circuitBreakerFailureRateThreshold(50.0f);

MultiClusterPooledConnectionProvider provider = new MultiClusterPooledConnectionProvider(builder.build());
```

Internally, the connection provider uses a [configurable circuit breaker and retry implementation](https://resilience4j.readme.io/docs/circuitbreaker) to determine when to fail over.

> In this configuration, we've set a sliding window size of 10 and a failure rate threshold of 50%. This means that a failover will be triggered if 5 out of any 10 calls to Redis fail.

Reviewer: I would add one more qualifier which is the sliding window type so they understand that there is a time component as well possible

Author: @aterleto See my suggested change in the next review.

> In this configuration, we've set a sliding window size of 10 and a failure rate threshold of 50%. This means that a failover will be triggered if 5 out of any 10 calls to Redis fail.

Author: ```suggestion
In this configuration, we've set a sliding window size of 10 and a failure rate threshold of 50%. This means that a failover will be triggered if 5 out of any 10 calls to Redis fail. Note that you can also configure a time-based sliding window. See the configuration details below for how to set this up.
